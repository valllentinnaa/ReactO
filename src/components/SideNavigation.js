import React from "react";

const SideNavigation = () => {
    return (
        <div>SideNavigation</div>
    );
};

export default SideNavigation;