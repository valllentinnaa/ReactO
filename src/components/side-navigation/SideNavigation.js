import React, {Component} from "react";
import UserContext from "../../Context";
import getNavigation from "../../utils/navigation";
import SideNavigationLink from "./SideNavigationLink";

class SideNavigation extends Component {

    static contextType = UserContext;

    render() {
        const {
            user
        } = this.context;

        const links = getNavigation(user);

        return (
            <div className="side-navigation">
                <div className="side-navigation-body">
                    {
                        links.map(navElement => {
                            return (
                                <SideNavigationLink
                                    key={navElement.title}
                                    href={navElement.link}
                                    title={navElement.title}
                                />
                            )
                        })
                    }
                </div>
            </div>
        )
    }
}

export default SideNavigation