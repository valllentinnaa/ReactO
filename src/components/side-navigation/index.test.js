import React from 'react'
import renderer from 'react-test-renderer'
import SideNavigation from './SideNavigation'
import TestingEnvironment from '../../test-utils/router'
import getNavigation from '../../utils/navigation'

describe('Navigation Component', () => {

    it('return authenticated routes from getNavigation', () => {
        const links = getNavigation({
            loggedIn: true,
            id: '123'
        });

        const authLinks = [
            {
                title: "Dashboard",
                link: "/dashboard"
            },
            {
                title: "Articles",
                link: "/articles"
            },
            {
                title: "Add your article",
                link: `/articles/new`
            },
            {
                "link": "/profile/123",
                "title": "Profile",
            },
        ]

        expect(links).toStrictEqual(authLinks)
    })

    it('should render authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    loggedIn: true,
                    id: '123'
                }
            }}>
                <SideNavigation/>
            </TestingEnvironment>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should render non-authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    loggedIn: false
                }
            }}>
                <SideNavigation/>
            </TestingEnvironment>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})