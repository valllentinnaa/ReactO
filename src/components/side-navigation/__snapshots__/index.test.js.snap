// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component should render authenticated routes 1`] = `
<div
  className="side-navigation"
>
  <div
    className="side-navigation-body"
  >
    <div
      className="side-navigation-item"
      data-test-id="link-Dashboard"
    >
      <a
        exact={true}
        href="/dashboard"
        onClick={[Function]}
      >
        Dashboard
      </a>
    </div>
    <div
      className="side-navigation-item"
      data-test-id="link-Articles"
    >
      <a
        exact={true}
        href="/articles"
        onClick={[Function]}
      >
        Articles
      </a>
    </div>
    <div
      className="side-navigation-item"
      data-test-id="link-Add your article"
    >
      <a
        exact={true}
        href="/articles/new"
        onClick={[Function]}
      >
        Add your article
      </a>
    </div>
    <div
      className="side-navigation-item"
      data-test-id="link-Profile"
    >
      <a
        exact={true}
        href="/profile/123"
        onClick={[Function]}
      >
        Profile
      </a>
    </div>
  </div>
</div>
`;

exports[`Navigation Component should render non-authenticated routes 1`] = `
<div
  className="side-navigation"
>
  <div
    className="side-navigation-body"
  >
    <div
      className="side-navigation-item"
      data-test-id="link-Dashboard"
    >
      <a
        exact={true}
        href="/dashboard"
        onClick={[Function]}
      >
        Dashboard
      </a>
    </div>
    <div
      className="side-navigation-item"
      data-test-id="link-Price Plans"
    >
      <a
        exact={true}
        href="/plans"
        onClick={[Function]}
      >
        Price Plans
      </a>
    </div>
  </div>
</div>
`;
